src/main/scala/com/vngrs/etl/Loader.scala

```diff
@@ -0,0 +1,31 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+trait Loader[A] {
+
+  protected[etl] def apply(input: RDD[A]): Loader[A]
+
+}
+
+object Loader {
+  def apply[A](f: RDD[A] => Unit): Loader[A] = new Loader[A]() {
+    override def apply(data: RDD[A]): Loader[A] = {
+      f(data)
+      this
+    }
+  }
+
+  def file(path: String): Loader[String] = new Loader[String] {
```

Comment:
It feels like we can define this functionality on type level.

Instead of creating a new method for every different loader, we need to extend Loader class with necessarry behaviour.

This way, user of this library can easily defin functionality for their custom Loaders/Extractrors/Transformers.

For instance:
```
trait Loader[A] {
protected[etl] abstract def apply(input: RDD[A]): Loader[A]
}

case class FileLoader extends Loader {
def apply() = ...
}

case class MySqlLoader extends Loader {
def apply() = ...
}
``` 
I don't know how to handle generic typing or custom constructors for every Loader implementation though